src/hdmf/common/table.py

```diff
@@ -1421,6 +1421,26 @@ def __repr__(self):
                                                               id(self.table))
         return template
 
+    def _validate_on_set_parent(self):
+        # when this DynamicTableRegion is added to a parent, check:
+        # 1) if the table was read from a written file, no need to validate further
+        p = self.table
+        while p is not None:
+            if p.container_source is not None:
+                return super()._validate_on_set_parent()
+            p = p.parent
+
+        # 2) if none of the ancestors are ancestors of the linked-to table, then when this is written, the table
+        # field will point to a table that is not in the file
+        table_ancestor_ids = [id(x) for x in self.table.get_ancestors()]
+        self_ancestor_ids = [id(x) for x in self.get_ancestors()]
+
+        if set(table_ancestor_ids).isdisjoint(self_ancestor_ids):
```

Comment:
This will check to see of the table and the DTR have some shared ancestor, aka check that its in the same file. If not then it'll be an external link and a warning will be thrown but it will be allowed to write. Correct?